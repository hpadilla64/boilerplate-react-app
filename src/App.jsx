import React from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Button, Icon } from "antd";
import {
  numberWithCommas,
  debounce,
  jsonToURL,
  slugify,
  isMobile
} from "./utils/Utils";
import interfaceActions from "./actions/UserInterface";

const App = props => {
  const { ui, addCount } = props;
  console.log(process.env.NODE_ENV);

  const testUtils = debounce(() => {
    console.log("debounce!");
    console.log(
      jsonToURL({
        prop: "asdfg",
        arr: [123, 456]
      })
    );
    console.log(slugify("Méx and double ñ"));
    console.log("is mobile? ", isMobile());
  }, 500);

  return (
    <div>
      <header>
        <img
          src="./styles/images/logo.png"
          alt=""
          style={{
            width: "5em",
            height: "auto"
          }}
        />
      </header>
      <p>
        <Icon type="exclamation-circle" /> Here is the app base container
      </p>
      <p>Clicks: {numberWithCommas(ui.count)}</p>
      <Button type="primary" onClick={addCount}>
        Aumentar click
      </Button>
      <br />
      <Button onClick={testUtils}>Test utils</Button>
    </div>
  );
};

const mapStateToProps = state => {
  return {
    ui: state.UserInterface
  };
};

const mapDispatchToProps = dispatch => {
  const { addCount } = interfaceActions.creators;
  return bindActionCreators(
    {
      addCount
    },
    dispatch
  );
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
