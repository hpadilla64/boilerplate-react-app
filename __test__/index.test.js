import React from "react";
import ReactDOM from "react-dom";
import App from "../src/App";

jest.mock("react-dom", () => ({ render: jest.fn() }));

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  global.document.getElementById = id => id === "app-container" && div;
  expect(ReactDOM.render).toMatchSnapshot();
});
