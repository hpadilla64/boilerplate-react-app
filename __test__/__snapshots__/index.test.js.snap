// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      <App />,
      <div />,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
